addons/pos_sale/tests/test_pos_sale_flow.py

```diff
@@ -540,3 +540,38 @@ def test_untaxed_invoiced_amount(self):
         self.env['pos.order'].create_from_ui([pos_order])
         self.assertEqual(sale_order.order_line[0].untaxed_amount_invoiced, 10, "Untaxed invoiced amount should be 10")
         self.assertEqual(sale_order.order_line[1].untaxed_amount_invoiced, 0, "Untaxed invoiced amount should be 0")
+
+    def test_pos_user_load_order(self):
+        product_a = self.env['product.product'].create({
+            'name': 'Product A',
+            'available_in_pos': False,
+            'type': 'product',
+            'lst_price': 10.0,
+            'taxes_id': [],
+        })
+
+        partner_test = self.env['res.partner'].create({'name': 'Test Partner'})
+
+        sale_order = self.env['sale.order'].create({
+            'partner_id': partner_test.id,
+            'order_line': [(0, 0, {
+                'product_id': product_a.id,
+                'name': product_a.name,
+                'product_uom_qty': 1,
+                'product_uom': product_a.uom_id.id,
+                'price_unit': product_a.lst_price,
+            })],
+        })
+        sale_order.action_confirm()
+
+        self.pos_user.write({
+            'groups_id': [
+                (4, self.env.ref('stock.group_stock_user').id),
+                (4, self.env.ref('sales_team.group_sale_salesman_all_leads').id),
+            ]
+        })
```

Comment:
may I know why you are adding rights to the pos_user? I thought the whole purpose of the fix is to allow them to have less rights and ignore the warning.

Reply:
It needs to have access to the sale order.